lib/node_modules/@stdlib/math/base/special/csignumf/lib/assign.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// MODULES //
+
+var Complex64 = require( '@stdlib/complex/float32/ctor' );
+var float64ToFloat32 = require( '@stdlib/number/float64/base/to-float32' );
+var real = require( '@stdlib/complex/float32/real' );
+var imag = require( '@stdlib/complex/float32/imag' );
+var csignumf = require( './main.js' );
+
+
+// MAIN //
+
+/**
+* Evaluates the signum function for a single-precision complex number and assigns results to a strided output array.
+*
+* @param {number} re - real component
+* @param {number} im - imaginary component
+* @param {Collection} out - output array
+* @param {integer} strideOut - stride length for output array
+* @param {NonNegativeInteger} offsetOut - starting index for output array
+* @returns {Collection} output array
+*
+* @example
+* var Float32Array = require( '@stdlib/array/float32' );
+*
+* var out = assign( 3.0, -4.0, new Float32Array( 2 ), 1, 0 );
+* // returns <Float32Array>[ 0.6000000238418579, -0.800000011920929 ]
+*/
+function assign( re, im, out, strideOut, offsetOut ) {
+	var z = new Complex64( re, im );
+	var v = csignumf( z );
+
+	re = float64ToFloat32( re );
+	im = float64ToFloat32( im );
+
+	out[ offsetOut ] = real( v );
+	out[ offsetOut + strideOut ] = imag( v );
+
+	return out;
```

Comment:
```suggestion
	var z;
	var v;

	z = new Complex64( float64ToFloat32( re ), float64ToFloat32( im ) );
	v = csignumf( z );
	out[ offsetOut ] = real( v );
	out[ offsetOut+strideOut ] = imag( v );
	return out;
```

You can also replace `float64ToFloat32` with `f32`